zkvm/src/vm.rs

```diff
@@ -672,20 +673,26 @@ where
         Ok(())
     }
 
+
     fn add_range_proof(&mut self, bitrange: BitRange, expr: Expression) -> Result<(), VMError> {
-        let (lc, assignment) = match expr {
-            Expression::Constant(x) => (r1cs::LinearCombination::from(x), Some(x)),
+        match expr {
+            Expression::Constant(x) => {
+                if x.in_range() {
+                    Ok(())
+                } else {
+                    Err(VMError::InvalidBitrange))
+                }
+            },
             Expression::LinearCombination(terms, assignment) => {
-                (r1cs::LinearCombination::from_iter(terms), assignment)
+                spacesuit::range_proof(
+                    self.delegate.cs(),
+                    r1cs::LinearCombination::from_iter(terms),
+                    ScalarWitness::option_to_integer(assignment)?,
+                    bitrange,
```

Comment:
I'd hard-code 64-bit range here, by passing `BitRange::new(64)` and removing `:n` parameter from the `range` instruction. This would require update to the spec too, and the according tweaks in program.rs and ops.rs